app/controllers/network_router_controller.rb

```diff
@@ -12,4 +12,259 @@ class NetworkRouterController < ApplicationController
+
+      add_flash(_("Network Router creation failed: Task start failed: ID [%{id}]") %
+                {:id => task_id.inspect}, :error) unless task_id.kind_of?(Fixnum)
+
+      if @flash_array
+        javascript_flash(:spinner_off => true)
+      else
+        initiate_wait_for_task(:task_id => task_id, :action => "create_finished")
+      end
+    end
+  end
+
+  def create_finished
+    task_id = session[:async][:params][:task_id]
+    router_name = session[:async][:params][:name]
+    task = MiqTask.find(task_id)
+    if MiqTask.status_ok?(task.status)
+      add_flash(_("%{model} \"%{name}\" created") % { :model => ui_lookup(:table => 'network_router'),
+                                                      :name  => router_name })
+    else
+      add_flash(
+        _("Unable to create %{model} \"%{name}\": %{details}") % { :model   => ui_lookup(:table => 'network_router'),
+                                                                   :name    => router_name,
+                                                                   :details => task.message }, :error)
+    end
+
+    @breadcrumbs.pop if @breadcrumbs
+    session[:edit] = nil
+    session[:flash_msgs] = @flash_array.dup if @flash_array
+
+    javascript_redirect :action => "show_list"
+  end
+
+  def delete_network_routers
+    assert_privileges("network_router_delete")
+
+    routers = if @lastaction == "show_list" || (@lastaction == "show" && @layout != "network_router")
+                find_checked_items
+              else
+                [params[:id]]
+              end
+
+    if routers.empty?
+      add_flash(_("No router were selected for deletion.") % {
+        :models => ui_lookup(:tables => "network_router")
+      }, :error)
+    end
+
+    routers_to_delete = []
+    routers.each do |s|
+      router = NetworkRouter.find_by_id(s)
+      if router.nil?
+        add_flash(_("Router no longer exists.") % {:model => ui_lookup(:table => "network_router")}, :error)
+      else
+        routers_to_delete.push(router)
+      end
+    end
+    process_network_routers(routers_to_delete, "destroy") unless routers_to_delete.empty?
+
+    # refresh the list if applicable
+    if @lastaction == "show_list"
+      show_list
+      @refresh_partial = "layouts/gtl"
+    elsif @lastaction == "show" && @layout == "network_router"
+      @single_delete = true unless flash_errors?
+      if @flash_array.nil?
+        add_flash(_("The selected Router was deleted") % {:model => ui_lookup(:table => "network_router")})
+      end
+    end
+  end
+
+  def edit
+    assert_privileges("network_router_edit")
+    @router = find_by_id_filtered(NetworkRouter, params[:id])
+    @in_a_form = true
+    drop_breadcrumb(
+      :name => _("Edit Router \"%{name}\"") % {:model => ui_lookup(:table => 'network_router'), :name => @router.name},
+      :url  => "/network_router/edit/#{@router.id}"
+    )
+  end
+
+  def get_checked_router_id(params)
```

Comment:
Good question but apparently the called method #find_checked_item is using params, I got inspired from existing code.